kyo-data/shared/src/main/scala/kyo/Maybe.scala

```diff
@@ -16,6 +16,7 @@ export Maybe.Present
 /** Companion object for Maybe type */
 object Maybe:
     inline given [A, B](using inline ce: CanEqual[A, B]): CanEqual[Maybe[A], Maybe[B]] = CanEqual.derived
+    inline given [A: Flat]: Flat[Maybe[A]]                                             = Flat.unsafe.bypass
```

Comment:
The workaround hardcoding the type in the macro was incorrect. A `Maybe[A]` can only be considered flat if `A` is flat type given that the internal representation is flat type union.